sentry-android-timber/src/main/java/io/sentry/android/timber/SentryTimberTree.kt

```diff
@@ -0,0 +1,74 @@
+package io.sentry.android.timber
+
+import android.util.Log
+import io.sentry.core.IHub
+import io.sentry.core.SentryEvent
+import io.sentry.core.SentryLevel
+import io.sentry.core.protocol.Message
+import timber.log.Timber
+
+/**
+ * Sentry Timber tree which is responsible to capture events via Timber
+ */
+class SentryTimberTree(private val hub: IHub, private val minLevel: SentryLevel) : Timber.Tree() {
+
+    /**
+     * Compares the minLevel given by the user and the level given by the event
+     * to capture an event or not.
+     */
+    override fun isLoggable(tag: String?, priority: Int): Boolean {
+        val sentryLevel = getSentryLevel(priority)
+
+        // do not log if it's lower than min. required level.
+        return sentryLevel.ordinal >= minLevel.ordinal
+    }
+
+    /**
+     * Captures a Sentry Event if the min. level is equal or higher than the min. required level.
+     */
+    override fun log(priority: Int, tag: String?, message: String, throwable: Throwable?) {
+        val sentryEvent = SentryEvent()
+        sentryEvent.level = getSentryLevel(priority)
+
+        // if there's no throwable, should it be a breadcrumb then?
+        throwable?.let {
+            sentryEvent.setThrowable(it)
+        }
+
+        // timber concatenates the stacktrace to the exception if the exception is not null
+        // let's take only the 1st line as this is probably the only message written by the user.
+        // or the 1st line of the exception which is probably the same as event.title
+        // eg java.lang.Exception: java.lang.Exception: java.lang.Exception: Some exception.
+        // timber should offer a way of disabling it
+        val realMessage = message.splitToSequence(delimiters = *arrayOf("\n"))
```

Comment:
`h`: I think if the user calls `Timber.e("message\nhello\nworld")` this code leads to cutting off the `hello\nworld`. I don't know if it is a good idea to implement this feature here like this. I would just take the message as it is. When Timber for some reason changes its implementation we also need to adapt this functionality here.

Reply:
I do agree, but it's the only workaround I could find to not log the whole exception stack trace, when Proguard is enabled, it's just useless and I could not find a delimiter that separates a real message to the exception

Reply:
As already pointed out I would just take the message from Timber and not cut it off. This is how Timber is implemented and if we keep the cutting of users are going to lose their information after the first `\n`.

Reply:
@philipphofmann this will have the side effect of people thinking that is us not demangling the stack traces properly.
I'd like a 3rd vote here, both options are suboptimal.

Reply:
I would also go with not splitting this and just use the message as it is.

Reasoning: We rather send too much than too little. This could have consequences that things a user wants to have for sure are not in the message.